cmake/flags.cmake

```diff
@@ -97,6 +97,7 @@ function(setupBuildFlags)
       )
 
       set(osquery_linux_common_link_options
+        -static-libstdc++
```

Comment:
This should be used only if we don't build with the custom toolchain, otherwise it brings in another C++ library plus other GCC libs that we don't want.
Also, another couple of things: this should be set to third party libraries too and, why static? 
I think that if someone builds with the system compiler it's probably targeting the same distro/platform, and in general packages for a distro prefer to use the shared version.